lib/rules/ngInject-angular-classes.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const utils = require('./../utils/utils');
+
+const prologueDirective = 'ngInject';
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "Should use ngInject",
+      category: "Possible Errors",
+      recommended: true
+    },
+    schema: []
+  },
+  create: (context) => {
+    return {
+      'ClassDeclaration': (node) => {
+        if (utils.isClassEmpty(node)) {
+          return;
+        }
+        const constructor = utils.getConstructor(node);
+        if (!constructor || constructor.value.params.length === 0) {
+          return
+        }
+        if (utils.classConstructorContainsDirectiveProlog(constructor, prologueDirective)) {
+          return;
+        }
```

Comment:
From this point on I would like more clarity as to what each of these tests is doing. Comments are ok, as are named functions. But at present I dont get it ?